### ISSUE TITLE
WebGLPointsLayer slows when using larger texture maps

### ISSUE THREAD

**Issue by jellyedwards** [openlayers/openlayers#10664]
**Describe the bug**
The WebGLPointsLayer sprite demo has a small icon set with a png size of 128 x 64. If you need many more icons and your png is larger, say 2048 x 2048 in my case, performance noticeably dips:

![10fpsWebGL](https://user-images.githubusercontent.com/14051992/74529773-734a0500-4f20-11ea-8e3c-df78333f48db.gif)

**To Reproduce**
1. Change the image in the sprite demo to a much larger texture
2. Run the demo and for Chrome I go into developer tools (F12)->Console, hit the vertical ellipsis and select Rendering, then enable the FPS counter:
![image](https://user-images.githubusercontent.com/14051992/74529937-cd4aca80-4f20-11ea-9872-b8bfef4ffddc.png)
3. You should see a low FPS (< 60FPS) as in the image above.

**Expected behavior**
FPS shouldn't be so harshly affected by a bigger texture (in my case it's still < 1MB).

I believe this is because webgl\Helper.js reloads the image onto the graphics card for every frame (line 557 onwards):

```javascript
      // apply value based on type
      if (value instanceof HTMLCanvasElement || value instanceof HTMLImageElement || value instanceof ImageData) {
        // create a texture & put data
        if (!uniform.texture) {
          uniform.texture = gl.createTexture();
        }
        gl.activeTexture(gl[`TEXTURE${textureSlot}`]);
        gl.bindTexture(gl.TEXTURE_2D, uniform.texture);
        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR);
        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);

        const imageReady = !(value instanceof HTMLImageElement) || /** @type {HTMLImageElement} */(value).complete;
        if (imageReady) {
          gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, value);
        }

        // fill texture slots by increasing index
        gl.uniform1i(this.getUniformLocation(uniform.name), textureSlot++);

      }
```

I've tried a simple only-reload-if-different approach:

```javascript
...
        // create a texture & put data
        if (!uniform.texture) {
          uniform.prevValue = undefined;
          uniform.texture = gl.createTexture();
        }
...
        const imageReady = !(value instanceof HTMLImageElement) || /** @type {HTMLImageElement} */(value).complete;
        // only reload the image if it's changed
        if (imageReady && uniform.prevValue !== value) {
          uniform.prevValue = value;
          gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, value);
        }
...
```

And FPS is at 60 with the larger texture:
![60fpsWebGL](https://user-images.githubusercontent.com/14051992/74530179-5f52d300-4f21-11ea-9099-ea20b883f3b9.gif)

**Comment by jahow** [comment]
Nice! Would you be able to provide a pull request for this improvement? Thanks for investigating

**Comment by jellyedwards** [comment]
Sure, no problem

**Comment by fredj** [comment]
fixed with #10684
